spec/routes_spec.rb

```diff
@@ -0,0 +1,24 @@
+routes = Rails.application.routes.routes.each_with_object({}) do |route, obj|
+  controller, action = route.defaults.values_at(:controller, :action)
+
+  next if controller.nil? || controller.starts_with?('api/') || controller.starts_with?('rails/')
+
+  klass = "#{controller}_controller".camelize.constantize
+
+  obj[klass] ||= []
+  obj[klass] << action
+end
```

Comment:
It might be useful to move this into a method in spec/support so it can be reused / cached, but that can be a follow up PR.